Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2328 -- Clickable Subject chips in Policy Repo results

Resolves [EREGCSC-2328](https://jiraent.cms.gov/browse/EREGCSC-2328)

**Description**

When you click a subject chip in the Policy Repository results list, that subject and only that subject is selected. 

No state is preserved – no subject or search query state.  Every filter parameter is cleared except for the selected subject.

**This pull request changes:**

- Makes subject chips clickable and uses the hover styles from the Resources page chips
- Clears `q`, `subjects`, and `type` parameters on click.  Resets `type` to `all` and makes the clicked subject the only subject selected
- Refactors document type selector logic to better handle Forward/Back actions by listening to `onpopstate`
- Further refactors document type selector logic to use an `onclick` event to dispatch `router` event
   - No longer `watch`ing the `checkedBoxes` ref to dispatch that event
   - `checkedBoxes` ref and `v-model` are now only used to control the display state of the checkboxes

**Steps to manually verify this change:**

1. Visit [experimental deployment](https://mywdnmntwk.execute-api.us-east-1.amazonaws.com/dev1017/policy-repository)
2. De-select "Public" checkbox so that only the "Internal" checkbox is selected
3. Search for "raisin"
4. Add "ABP" subject from the Subject Selector list in the left sidebar
5. Click the "Access to Services" subject chip.  
   - Notice that the search input should be cleared and the `q` query param removed from the URL
   - Notice that both checkboxes are back to being checked
   - Notice that only the clicked subject "Access to Services" is selected.  "ABP" should no longer be selected
